be/src/cloud/cloud_cumulative_compaction.cpp

```diff
@@ -48,7 +48,8 @@ CloudCumulativeCompaction::CloudCumulativeCompaction(CloudStorageEngine& engine,
 CloudCumulativeCompaction::~CloudCumulativeCompaction() = default;
 
 Status CloudCumulativeCompaction::prepare_compact() {
-    if (_tablet->tablet_state() != TABLET_RUNNING) {
+    if (_tablet->tablet_state() != TABLET_RUNNING &&
+        dynamic_cast<CloudTablet*>(_tablet.get())->alter_version() == -1) {
```

Comment:
While waiting for the waterline, the new tablet still cannot do compaction?

Reply:
_tablet->tablet_state() is always not equals TABLET_RUNNING until the schema change done. So this condition is not concerned with schema change job intermediate state.